src/test/java/io/aiven/kafka/connect/gcs/GcsSinkTaskTest.java

```diff
@@ -71,6 +77,58 @@
         createRecord("topic0", 2, "key9", "value9", 51, 1009)
     );
 
+    private List<Record> createTestRecords() {
+        final List<Record> records = new ArrayList<>();
+        records.add(Record.of("topic0", "key0", "value0", 0, 10, 1000, createHeaders()));
+        records.add(Record.of("topic0", "key1", "value1", 1, 20, 1001, createHeaders()));
+        records.add(Record.of("topic1", "key2", "value2", 0, 30, 1002, createHeaders()));
+        records.add(Record.of("topic1", "key3", "value3", 1, 40, 1003, createHeaders()));
+        records.add(Record.of("topic0", "key4", "value4", 2, 50, 1004, createHeaders()));
+
+        records.add(Record.of("topic0", "key5", "value5", 0, 11, 1005, createHeaders()));
+        records.add(Record.of("topic0", "key6", "value6", 1, 21, 1006, createHeaders()));
+        records.add(Record.of("topic1", "key7", "value7", 0, 31, 1007, createHeaders()));
+        records.add(Record.of("topic1", "key8", "value8", 1, 41, 1008, createHeaders()));
+        records.add(Record.of("topic0", "key9", "value9", 2, 51, 1009, createHeaders()));
+        return records;
+    }
+
+    private Map<String, Collection<Record>> toBlobNameWithRecordsMap(final String compression,
+                                                                     final List<Record> records) {
+        final CompressionType compressionType = CompressionType.forName(compression);
+        final String extension = compressionType.extension();
+        final Map<String, Integer> topicPartitionMinimumOffset = new HashMap<>();
+        final Map<String, Collection<Record>> blobNameWithRecordsMap = new HashMap<>();
+        for (final Record record : records) {
+            final String key = record.topic + "-" + record.partition;
+            final int offset = record.offset;
+            topicPartitionMinimumOffset.putIfAbsent(key, offset);
+            if (topicPartitionMinimumOffset.get(key) > offset) {
+                topicPartitionMinimumOffset.put(key, offset);
+            }
+        }
+        for (final Record record : records) {
+            final int offset = topicPartitionMinimumOffset.get(record.topic + "-" + record.partition);
+            final String key = record.topic + "-" + record.partition + "-" + offset + extension;
+            blobNameWithRecordsMap.putIfAbsent(key, new ArrayList<>());
+            blobNameWithRecordsMap.get(key).add(record);
+        }
+        return blobNameWithRecordsMap;
+    }
+
+    private List<SinkRecord> toSinkRecords(final List<Record> records) {
+        final List<SinkRecord> sinkRecords = new ArrayList<>();
+        for (final Record record : records) {
+            sinkRecords.add(toSinkRecord(record));
+        }
+        return sinkRecords;
```

Comment:
FWIW, but maybe you consider this simpler
```suggestion
        return records.stream().map(this::toSinkRecord).collect(Collectors.toList());
```